src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -263,7 +261,14 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
-							if (t.init(res.exprs, i, getParser().getHasDelayBefore(), res)) {
+							List<TriggerSection> sections = getParser().getCurrentSections();
+							TriggerSection current = sections.isEmpty() ? null : sections.get(sections.size() - 1);
+							if (t.getUsableEvents().length > 0 && !CollectionUtils.containsAny(t.getUsableEvents(), getParser().getCurrentEvents()))
+								Skript.error("You cannot use this element in a(n) " + getParser().getCurrentEventName() + " event.");
```

Comment:
```suggestion
								Skript.error("You cannot use this element in " + Utils.a(getParser().getCurrentEventName()) + " event.");
```
This will add the indefinite article by itself

Reply:
Ah okay thank you, shall add this